### ISSUE TITLE
inline assembly warnings are treated as errors

### ISSUE THREAD

**Issue by llvmbot** [llvm/llvm-project#9356]
|  |  |
| --- | --- |
| Bugzilla Link | [8984](https://llvm.org/bz8984) |
| Version | trunk |
| OS | All |
| Reporter | LLVM Bugzilla Contributor |
| CC | @Arnaud-de-Grandmaison-ARM |

## Extended Description 
A simple test case with this unimplemented CFI directive:

$ cat inline-asm.c

asm(".cfi_adjust_cfa_offset 4\n\t");

produces this:

$ clang -c inline-asm.c
<inline asm>:1:1: error: warning: ignoring directive for now
.cfi_adjust_cfa_offset 4
^
1 error generated.

It's clearly a warning, yet clang reports it as an error. Aside from the obvious point that someone needs to finish up CFI support, I don't think that warnings should be reported as errors unless -Werror gets passed to the compiler.

**Comment by gwelymernans** [comment]
This looks to be fixed?

```
$ clang -c pr.c
pr.c:2:5: warning: scale factor without index register is ignored [-Winline-asm]
asm("movl (%rax,4), %eax");
    ^
<inline asm>:1:13: note: instantiated into assembly here
        movl (%rax,4), %eax
                   ^
1 warning generated.
```
